code/game/machinery/computer/ai_core.dm

```diff
@@ -30,48 +43,43 @@
 					to_chat(user, "The welder must be on for this task.")
 					return
 				playsound(loc, 'sound/items/Welder.ogg', 50, 1)
-				if(do_after(user, src, 20))
-					if(!src || !WT.remove_fuel(0, user))
+				if(do_after(user, src, 2 SECONDS))
+					if(!src || state != NOCIRCUITBOARD || !WT.remove_fuel(0, user))
 						return
 					to_chat(user, "<span class='notice'>You deconstruct the frame.</span>")
-					new /obj/item/stack/sheet/plasteel( loc, 4)
+					drop_stack(/obj/item/stack/sheet/plasteel, loc, 4, user)
 					qdel(src)
 			if(istype(P, /obj/item/weapon/circuitboard/aicore) && !circuit)
 				if(user.drop_item(P, src))
 					playsound(loc, 'sound/items/Deconstruct.ogg', 50, 1)
 					to_chat(user, "<span class='notice'>You place the circuit board inside the frame.</span>")
-					icon_state = "1"
 					circuit = P
 					state = UNSECURED_CIRCUITBOARD
 		if(UNSECURED_CIRCUITBOARD)
 			if(isscrewdriver(P) && circuit)
 				playsound(loc, 'sound/items/Screwdriver.ogg', 50, 1)
 				to_chat(user, "<span class='notice'>You screw the circuit board into place.</span>")
 				state = SECURED_CIRCUITBOARD
-				icon_state = "2"
 			if(iscrowbar(P) && circuit)
 				playsound(loc, 'sound/items/Crowbar.ogg', 50, 1)
 				to_chat(user, "<span class='notice'>You remove the circuit board.</span>")
 				state = NOCIRCUITBOARD
-				icon_state = "0"
 				circuit.forceMove(loc)
 				circuit = null
 		if(SECURED_CIRCUITBOARD)
 			if(isscrewdriver(P) && circuit)
 				playsound(loc, 'sound/items/Screwdriver.ogg', 50, 1)
 				to_chat(user, "<span class='notice'>You unfasten the circuit board.</span>")
 				state = UNSECURED_CIRCUITBOARD
-				icon_state = "1"
-			if(istype(P, /obj/item/stack/cable_coil))
-				if(P:amount >= 5)
+			var/obj/item/stack/cable_coil/cable_coil = P
+			if(istype(cable_coil))
```

Comment:
if(iscoil(P))

Reply:
Not that you're incorrect, but `P` needs to be cast as `/obj/item/stack/cable_coil` anyway in order to access its `amount` var.